<h1 id="mars-malaria-resistance-surveillance">MaRS (Malaria Resistance Surveillance)</h1> <p>The emergence of resistance to all currently available antimalarial drugs in multiple regions of the world represents a current global public health challenge. In order to monitor and address this situation, faster and more effective surveillance tools are required to track and monitor the emergence and evolution of drug resistance in malaria. The Malaria Resistance Surveillance (MaRS) project aims to address this challenge by collating and mapping genetic polymorphisms associated with drug resistance in malaria around the world. The project achieves this by employing a targeted amplicon deep sequencing (TADS) approach <a href="https://github.com/CDCgov/MaRS/tree/master/lab_sop">Lab Protocol</a> to detect single nucleotide polymorphisms on all major malaria drug resistance genes associated genes in samples sourced from travelers returning to the US from overseas, as well as samples actively collected in collaboration with partners from other countries.</p> <p>Data for this project can be found at the following link <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA428490">NCBI BioProject</a>. Collaborators are encouraged to submit their own data using this <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA428490">NCBI BioProject</a></p> <p>The Malaria Resistance Surveillance or MaRS analysis pipline, is an attempt at standardizing the workflow for identifying both known and new polymorhisms in <em>P.falciparum</em> genes associated with drug resistance.</p> <h2 id="version-histroy">Version Histroy</h2> <ul> <li>Version 1.1.1 (1/9/2018) <ul> <li>Fastq samples detected from directory</li> <li>Reference and dependencies for analysis on P.falciparum provided with the git bundle</li> <li>BBDuk used to trim reads</li> <li>BWA used for read alignment</li> <li>Variant calling using Samtools and GATK</li> <li>Summary tables for samples with separate tables for known and novel variants</li> <li>Summary heatmaps and frequency graphs generated</li> <li>The framework has been tested to work on Ubuntu 16.04 and RedHat Enterprise Linux Server Edition 6.8</li> </ul> </li> </ul> <h1 id="setup">Setup</h1> <h2 id="getting-started">Getting started</h2> <ol> <li>Download git repository:</li> </ol> <p>Clone the master branch of this repository.</p> <pre><code class="language-{sh}">git clone https://github.com/CDCgov/MaRS.git
</code></pre> <ol> <li>Download dependencies:</li> </ol> <p>Download dependencies listed below. All the tools that are required to run MaRS included in the lib directory of the repository. However, make sure that the Python and Java versions and packages are installed on the system.</p> <ol> <li>Your first analysis:</li> </ol> <p>Follow the directory structure listed below and use the run script included with the bundle to run your first analysis.</p> <pre><code class="language-{sh}">sh run.sh &lt;path to experiment folder&gt; &lt;path to output folder&gt;
</code></pre> <p>For example if you have stored your fastq files in <code class="highlighter-rouge">fq/</code> folder and you want to store the results in the folder <code class="highlighter-rouge">local/</code>. You can run the following command from the MaRS directory.</p> <pre><code class="language-{sh}">sh run.sh fq/ local/
</code></pre> <h2 id="depdencies">Depdencies:</h2> <ol> <li><a href="https://www.python.org/download/releases/3.4.0/">Python3.4 </a></li> <li><a href="http://download.oracle.com/otn-pub/java/jdk/9.0.1+11/jre-9.0.1_linux-x64_bin.tar.gz">Java (Version : 9.0.1)</a></li> <li><a href="http://pandas.pydata.org/pandas-docs/stable/">Pandas (Version : 0.22.0)</a></li> <li><a href="https://www.scipy.org/install.html">Numpy (Version : 1.13.3)</a></li> <li><a href="https://seaborn.pydata.org/">Seaborn (Version : 0.8.1)</a></li> <li><a href="https://pypi.python.org/pypi/openpyxl">Openpyxl (Version : 2.4.9)</a></li> <li><a href="https://sourceforge.net/projects/bbmap/">BBMap (Version : v35.x)</a></li> <li><a href="http://bio-bwa.sourceforge.net/">BWA (Version : 0.7.12)</a></li> <li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2 (Version : 2.3.3.1)</a></li> <li><a href="http://snap.cs.berkeley.edu/">Snap (Version : 1.0beta23)</a></li> <li><a href="http://www.htslib.org/">Samtools (Version : 1.3.1)</a></li> <li><a href="http://www.htslib.org/">Bctools (Version : 1.3.1)</a></li> <li><a href="https://software.broadinstitute.org/gatk/download/">GATK (Version : 3.6-0-g89b7209)</a></li> </ol> <h2 id="directory-structure">Directory structure:</h2> <ol> <li>fq</li> </ol> <ul> <li>Contains all the input fastq files</li> </ul> <ol> <li>lib</li> </ol> <ul> <li>Contains the binaries for all the tools that MaRS can run</li> </ul> <ol> <li>local</li> </ol> <ul> <li>Local output folder</li> </ul> <ol> <li>pyamd</li> </ol> <ul> <li>Contains all the MaRS classes.</li> </ul> <ol> <li>ref</li> </ol> <ul> <li>Contains reference sequences. For the current version, the ref folder must also contain the alignment indicies.</li> </ul> <h2 id="creating-alignment-indicies">Creating alignment indicies:</h2> <ol> <li>BWA: <pre><code class="language-{sh}">bwa index &lt;reference-fasta.fa&gt;
</code></pre> </li> </ol> <h2 id="output-directory-structure">Output directory structure:</h2> <ul> <li> <p>For each sample</p> <ol> <li>CleanedFastq: Folder containing adapter trimmed and quality filtered fastq files</li> <li>output.sam : Contains reads aligned to reference genome</li> <li>output_sorted.bam : Sorted BAM containing aligned reads</li> <li>output_sorted_RG.bam : Read gtroup added BAM file</li> <li>outout_fixmate.bam : Final BAM file, with mate information corrected</li> <li>variants.vcf : Variant calls from samtools</li> <li>variants_gatk.vcf : Variant calls from GATK HaplotypeCaller</li> <li>variants_samtools.bed : Annotated variant calls from samtools, in tab delimited format</li> <li>variants_gatk.bed : Annotated variant calls from GATK, in tab delimited format</li> </ol> </li> <li> <p>For the study</p> <ol> <li>Study_variants.xlsx : Summary table of all known variants that confer drug resistance, for all samples in the study</li> <li>Study_depth.xlsx : Summary of depth of coverage for codon correponding to variants that confer drug resistance</li> <li>Study_al_freq.xlsx : Summary of allele frequency of variants that confer drug resistance</li> <li>Study_novel_exonic_variants.xlsx : Summary of all novel variants found in exonic regions, for all samples in the study</li> <li>Study_novel_intronic_variants.xlsx : Summary of all novel variants found in intronic regions, for all samples in the study</li> <li>Study_novel_var_af.xlsx : Summary of allele frequency of novel variants for all samples</li> <li>Study_novel_var_depth.xlsx : Summary of depth of coverage for codon correponding to novel variants for all samples</li> <li>voi_af_heatmap.jpg : Heatmap showing the allele frequency across all samples for variants known to confer drug resistance</li> <li>voi_dp_heatmap.jpg : Heatmap showing the codon coverage across all samples for variants known to confer drug resistance</li> <li>voi_frequency.jpg : Count plot showing frequency variants conferring drug resistance across all the samples in the study</li> </ol> </li> </ul>